src/main/java/gregtech/api/gui/widgets/PhantomSlotWidget.java

```diff
@@ -20,14 +22,27 @@
 
 public class PhantomSlotWidget extends SlotWidget implements IGhostIngredientTarget {
 
+    private boolean clearSlotOnRightClick;
+
     public PhantomSlotWidget(IItemHandlerModifiable itemHandler, int slotIndex, int xPosition, int yPosition) {
         super(itemHandler, slotIndex, xPosition, yPosition, false, true);
     }
 
+    public PhantomSlotWidget setClearSlotOnRightClick(boolean clearSlotOnRightClick) {
+        this.clearSlotOnRightClick = clearSlotOnRightClick;
+        return this;
+    }
+
     @Override
     public boolean mouseClicked(int mouseX, int mouseY, int button) {
         if (isMouseOverElement(mouseX, mouseY) && gui != null) {
-            gui.getModularUIGui().superMouseClicked(mouseX, mouseY, button);
+            if (button == 1 && clearSlotOnRightClick && !slotReference.getStack().isEmpty()) {
+                slotReference.putStack(ItemStack.EMPTY);
+                writeClientAction(2, buf -> {
```

Comment:
Should probably be put in GregtechDataCodes.

Reply:
Not a single widget does it via GregtechDataCodes and the widgets are trash anyways